### PR TITLE
[CWS] add fallback on /proc for ebpf-less fd resolution

### PR DESCRIPTION
**Pull request by paulcacheux**
### What does this PR do?

When running in wrap mode we should hopefully collect all file descriptor (or it's a bug), but when running in attach mode not having all fds in the cache is the regular mode of operation. To fight this this PR adds a fallback on /proc to query the latest fd path when it's not in the cache.

### CODE REVIEW

File: pkg/security/ptracer/fim_handlers.go [hunk 1]
```diff
@@ -250,6 +251,11 @@ func registerFIMHandlers(handlers map[int]syscallHandler) []string {
 			ShouldSend: isAcceptedRetval,
 			RetFunc:    nil,
 		},
+		{
+			ID:      syscallID{ID: Pipe2Nr, Name: "pipe2"},
```
Comment (spikat) [inline pkg/security/ptracer/fim_handlers.go:255]: shouldn't we handle `pipe` as well ? (even if not present on arm64)

File: pkg/security/ptracer/fim_handlers.go [hunk 2]
```diff
@@ -1193,14 +1206,34 @@ func handleFcntlRet(tracer *Tracer, process *Process, msg *ebpfless.SyscallMsg,
...
+
+	fd1 := int32(binary.NativeEndian.Uint32(fds[:sizeOfInt]))
+	fd2 := int32(binary.NativeEndian.Uint32(fds[sizeOfInt:]))
+
+	process.FdRes.Fd[fd1] = "pipe:"
+	process.FdRes.Fd[fd2] = "pipe:"
+
+	msg.Pipe = &ebpfless.PipeSyscallFakeMsg{
+		FdsPtr: tracer.ReadArgUint64(regs, 0),
+	}
```
Comment (spikat) [inline pkg/security/ptracer/fim_handlers.go:1233]: I don't think we need those lines

Comment (paulcacheux) [reply pkg/security/ptracer/fim_handlers.go:1233]: Ah yeah, sorry copy paste error

File: pkg/security/ptracer/fim_handlers.go (hunk 1 above)
Comment (paulcacheux) [reply pkg/security/ptracer/fim_handlers.go:255]: 👍

File: pkg/security/ptracer/fim_handlers.go [hunk 3]
```diff
@@ -1193,14 +1206,34 @@ func handleFcntlRet(tracer *Tracer, process *Process, msg *ebpfless.SyscallMsg,
...
 			process.FdRes.Fd[int32(ret)] = path
 		}
 	}
 	return nil
 }
 
+const sizeOfInt = 4
+
+func handlePipe2Ret(tracer *Tracer, process *Process, msg *ebpfless.SyscallMsg, regs syscall.PtraceRegs, _ bool) error {
+	fds, err := tracer.readData(process.Pid, msg.Pipe.FdsPtr, 2*sizeOfInt)
```
Comment (spikat) [inline pkg/security/ptracer/fim_handlers.go:1220]: We should start by checking the ret value here I guess, and exit if an error was returned

**Review by spikat — APPROVED** [review]
LGTM, thanks for the changes

**Pull request merged**